src/clientiface.h

```diff
@@ -201,10 +230,24 @@ class RemoteClient
 	//
 	u16 net_proto_version;
 
+	/* Authentication information */
+	std::string enc_pwd;
+	int chosen_mech;
+	void * auth_data;
+	u32 allowed_auth_mechs;
+	u32 allowed_sudo_mechs;
+
+	bool is_sudo_mech_allowed(int mech)
+	{ return allowed_sudo_mechs & mech; }
+	bool is_mech_allowed(int mech)
+	{ return allowed_auth_mechs & mech; }
+
 	RemoteClient():
 		peer_id(PEER_ID_INEXISTENT),
 		serialization_version(SER_FMT_VER_INVALID),
 		net_proto_version(0),
+		chosen_mech(0), // AUTH_MECHANISM_NONE,
```

Comment:
If this is initialized to AUTH_MECHANISM_NONE, then what's wrong with just setting it to AUTH_MECHANISM_NONE instead of 0?